pkg/assembler/backends/ent/backend/certifyVEXStatement.go

```diff
@@ -139,19 +140,30 @@ func (b *EntBackend) IngestVEXStatement(ctx context.Context, subject model.Packa
 }
 
 func (b *EntBackend) IngestVEXStatements(ctx context.Context, subjects model.PackageOrArtifactInputs, vulnerabilities []*model.VulnerabilityInputSpec, vexStatements []*model.VexStatementInputSpec) ([]string, error) {
-	var ids []string
+	var ids = make([]string, len(vexStatements))
+	eg, ctx := errgroup.WithContext(ctx)
```

Comment:
Suppose the IngestVEXStatement function receives a `ctx` from errgroup.WithContext(ctx), and this context gets canceled when any goroutine fails, could this potentially interrupt important operations inside IngestVEXStatement, such as database writes?


 Would it lead to data inconsistencies? 
 
 Would it be safer to use the original `ctx` for critical tasks in IngestVEXStatement and the derived context only for tasks that should be stopped if any goroutine fails to ensure critical operations aren't prematurely terminated?

Reply:
The "upsert" approach adopted in the Ent backend is meant to ensure idempotency to mutation endpoints to avoid any kind of data inconsistency or more complicated tasks management.

Reply:
Thank you! I wasn't aware of this. 👍